clash-lib/src/Clash/Driver.hs

```diff
@@ -466,86 +454,41 @@ generateHDL reprs domainConfs bindingsMap hdlState primMap tcm tupTcm typeTrans
 
       topTime <- hdlDocs `seq` Clock.getCurrentTime
       let edamFiles' = HashMap.insert (varUniq topEntity) (asEdamFile topNm <$> fileNames manifest') edamFiles
       return (topTime,manifest',seen2, edamFiles')
 
   -- Generate EDAM if needed
-  if opt_edalize opts
-    then let dir      = hdlDir </> maybe "" (const modNameS) annM
-             topFiles = fileNames manifest'
-             incFiles = concatMap (\u -> HashMap.lookupDefault [] u edamFiles') $ HashMap.lookupDefault [] (varUniq topEntity) deps
-             edamInfo = createEDAM topNm incFiles topFiles
-          in writeHDL dir ("edam.py", pprEdam edamInfo)
-
-    else pure ()
-
-  benchTime <- case benchM of
-    Just tb | not useCacheBench -> do
-      putStrLn $ "Clash: Compiling " ++ Data.Text.unpack (nameOcc (varName tb))
-
-      let modName' = Id.toText (lookupVarEnv' compNames tb)
-          hdlState2 = setModName modName' hdlState'
-
-      -- 1. Normalise testBench
-      let transformedBindings = normalizeEntity reprs bindingsMap primMap tcm tupTcm
-                                  typeTrans eval topEntityNames opts supplyTB tb
-      normTime <- transformedBindings `deepseq` Clock.getCurrentTime
-      let prepNormDiff = reportTimeDiff normTime topTime
-      putStrLn $ "Clash: Testbench normalization took " ++ prepNormDiff
-
-      -- 2. Generate netlist for topEntity
-
-      -- See [Note] Create HDL dir before netlist generation
-      let dir = hdlDir </> maybe "" t_name annM </> Data.Text.unpack modName'
-      prepareDir (opt_cleanhdl opts) (extension hdlState2) dir
-      -- Now start the netlist generation
-      (_topName, netlist, seen3) <-
-        genNetlist True opts reprs transformedBindings topEntityMap compNames primMap
-                   tcm typeTrans iw ite (SomeBackend hdlState') seen2 hdlDir prefixM tb
-
-      netlistTime <- netlist `deepseq` Clock.getCurrentTime
-      let normNetDiff = reportTimeDiff netlistTime normTime
-      putStrLn $ "Clash: Testbench netlist generation took " ++ normNetDiff
-
-      -- 3. Write HDL
-      let (hdlDocs,_,dfiles,mfiles) =
-            createHDL hdlState2 modName' seen3 netlist domainConfs Nothing
-                           (topNmT, Left manifest')
-      writeHDL (hdlDir </> maybe "" t_name annM) (head hdlDocs)
-      mapM_ (writeHDL dir) (tail hdlDocs)
-      copyDataFiles (opt_importPaths opts) dir dfiles
-      writeMemoryDataFiles dir mfiles
-
-      hdlDocs `seq` Clock.getCurrentTime
-
-    Just tb -> do
-      let tb' = Data.Text.unpack (nameOcc (varName tb))
-      putStrLn ("Clash: Compiling: " ++ tb')
-      putStrLn ("Clash: Using cached result for: " ++ tb')
-      return topTime
+  Monad.when (opt_edalize opts) $ do
+    let
+      topFiles = fileNames manifest'
+      incFiles = concatMap (\u -> HashMap.lookupDefault [] u edamFiles')
+               $ HashMap.lookupDefault [] (varUniq topEntity) deps
+      edamInfo = createEDAM topNm incFiles topFiles
```

Comment:
I would call it `createEdam` instead: no other definition uses EDAM all in capitals right now

Reply:
Ah alright, I didn't really touch that, just changed the indentation. I'll add a separate clean-up commit.

Reply:
O snap, that one's my fault then. Whoops...

Reply:
I'm going to refactor this in another PR if you don't mind.